test/plots/waterfallPlotTests.ts

```diff
@@ -1,123 +1,251 @@
-      let bars = renderArea.selectAll("rect")[0];
-      let data = dataset.data();
-      let yAccessor = plot.y().accessor;
-      let totalAccessor = plot.total().accessor;
-      for (let currentIndex = 1; currentIndex < bars.length; currentIndex++) {
-        let currentBar = d3.select(bars[currentIndex]);
-        let currentTotal = totalAccessor(data[currentIndex], currentIndex, dataset);
-        let currentValue = yAccessor(data[currentIndex], currentIndex, dataset);
-        let previousIndex = currentIndex - 1;
-        let previousBar = d3.select(bars[previousIndex]);
-        let previousTotal = totalAccessor(data[previousIndex], previousIndex, dataset);
-        let previousValue = yAccessor(data[previousIndex], previousIndex, dataset);
-        if (previousTotal) {
-          if (currentTotal || currentValue < 0) {
-            assert.isTrue(+previousBar.attr("y") === +currentBar.attr("y"), "bars are top/top aligned");
-          } else {
-            assert.closeTo(+previousBar.attr("y") - (+currentBar.attr("y") + +currentBar.attr("height")), 0, 1,
-              "bars are top/bottom aligned");
+      beforeEach(() => {
+        svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+        dataset = new Plottable.Dataset(growthBarData);
+        xScale = new Plottable.Scales.Category();
+        yScale = new Plottable.Scales.Linear();
+        plot = new Plottable.Plots.Waterfall<string, number>();
+        plot.x((d) => d.x, xScale);
+        plot.y((d) => d.y, yScale);
+        plot.total((d, i) => i === 0);
+        plot.addDataset(dataset);
+        plot.renderTo(svg);
+      });
+
+      it("classes growth bars", () => {
+        let bars = plot.content().selectAll("rect");
+        assert.strictEqual(bars.size(), growthBarData.length, "all bars are growth except for first");
+        bars.each(function(d, i) {
+          if (i === 0) {
+            return;
           }
-        } else {
-          if (previousValue > 0) {
-            if (currentTotal || currentValue < 0) {
-              assert.isTrue(+previousBar.attr("y") === +currentBar.attr("y"), "bars are top/top aligned");
-            } else {
-              assert.closeTo(+previousBar.attr("y") - (+currentBar.attr("y") + +currentBar.attr("height")), 0, 1,
-                "bars are top/bottom aligned");
-            }
-          } else {
-            if (currentTotal || currentValue < 0) {
-              assert.closeTo(+previousBar.attr("y") + +previousBar.attr("height") - +currentBar.attr("y"), 0, 1,
-                "bars are bottom/top aligned");
-            } else {
-              assert.closeTo((+previousBar.attr("y") + +previousBar.attr("height")) -
-                (+currentBar.attr("y") + +currentBar.attr("height")), 0, 1, "bars are bottom/bottom aligned");
-            }
+          let bar = d3.select(this);
+          assert.isTrue(bar.classed("waterfall-growth"), "bar classed as growth bar");
+        });
+        plot.destroy();
```

Comment:
I am making a habit to `destroy` every `Component` I create in a test, mainly since it is a very safe way to test it.  (Also has the nice benefit of cleaning up the test).